pkg/dataobj/tools/stats.go

```diff
@@ -69,3 +82,32 @@ func calculateSectionsPerTenantStats(s *Stats) {
 	}
 	s.SectionsPerTenantStats.P99 = float64(counts[idx])
 }
+
+// calculateSectionSizes calculates the median and other percentiles for the
+// size of sections.
+func calculateSectionSizeStats(s *Stats) {
```

Comment:
Guess you can mostly re-use `calculateSectionsPerTenantStats` for this?

Reply:
I created a generic function `calcPercentiles` to do this 👍